src/backend/aspen/workflows/nextclade/prep_samples.py

```diff
@@ -0,0 +1,119 @@
+
+
+@click.command("export")
+@click.option("sample_ids_fh", "--sample-ids-file", type=click.File("r"), required=True)
+@click.option("sequences_fh", "--sequences", type=click.File("w"), required=True)
+@click.option(
+    "pathogen_info_fh", "--pathogen-info-file", type=click.File("w"), required=True
+)
+def cli(
+    sample_ids_fh: io.TextIOBase,
+    sequences_fh: io.TextIOBase,
+    pathogen_info_fh: IO[str],
+):
+    """
+    Writes out a FASTA for the specified samples.
+
+    - sample_ids_fh: A file of sample ID primary keys, one ID per line.
+        Each sample must be for the same pathogen (all SARS-CoV-2, etc)
+    - sequences_fh: Output file to write FASTA for above samples.
+        NOTE Resulting FASTA will have its id lines (>) be those primary keys,
+        so anything that consumes these downstream results will be referring
+        to samples by PK, not by private/public identifier.
+    - pathogen_info_fh: Write out pathogen info for later use in workflow
+    """
+    sample_ids: list[int] = [int(id_line) for id_line in sample_ids_fh]
+    print("Fetching and writing FASTA for sample ids:", sample_ids)
+    interface: SqlAlchemyInterface = init_db(get_db_uri(Config()))
+    with session_scope(interface) as session:
+        all_samples: Iterable[Sample] = (
+            session.query(Sample)
+            .filter(Sample.id.in_(sample_ids))
+            .options(
+                joinedload(Sample.uploaded_pathogen_genome).undefer(
+                    UploadedPathogenGenome.sequence
+                ),
+            )
+        )
+
+        # Should pick it up as we run through samples. If missing at end, raise
+        pathogen_of_all_samples: Optional[Pathogen] = None
+        for sample in all_samples:
+
+            # `.pathogen` access is lazy, but only accesses once per pathogen
+            # type, so not an N+1 query issue where we need eager load.
+            pathogen: Pathogen = sample.pathogen
+            if pathogen_of_all_samples is None:
+                pathogen_of_all_samples = pathogen
+            # Safety: ensure all samples are same pathogen before Nextclade run
+            if pathogen != pathogen_of_all_samples:
+                err_msg = (
+                    f"ERROR -- Encountered differing pathogen types in list "
+                    f"of samples. Encountered both {pathogen.slug} and "
+                    f"{pathogen_of_all_samples.slug} in given samples. There "
+                    f"may also be others, this is just first difference found."
+                )
+                print(err_msg)
+                raise RuntimeError("Samples given are not all same pathogen")
+
+            uploaded_pathogen_genome = sample.uploaded_pathogen_genome
+            # Samples _should_ always have uploaded_pathogen_genome with
+            # sequence data on them in theory, but if it's missing, blow up.
+            if uploaded_pathogen_genome is None:
```

Comment:
if you add an `innerjoin=True` parameter to the joinedload above, you won't have to check for errors here.